var/spack/repos/builtin/packages/python/package.py

```diff
@@ -359,415 +331,78 @@ def flag_handler(self, name, flags):
-                config_args.append("--with-lto=thin")
+        # We need to be careful here. If the user is using an externally
+        # installed python, several different commands could be located
+        # in the same directory. Be as specific as possible. Search for:
+        #
+        # * python3.6
+        # * python3
+        # * python
+        #
+        # in that order if using python@3.6.5, for example.
+        version = self.spec.version
+        for ver in [version.up_to(2), version.up_to(1), ""]:
+            if not sys.platform == "win32":
+                path = os.path.join(self.prefix.bin, "python{0}".format(ver))
             else:
-                config_args.append("--with-lto")
-            config_args.append("--with-computed-gotos")
-
-        if spec.satisfies("@3.7 %intel"):
-            config_args.append("--with-icc={0}".format(spack_cc))
-
-        if "+debug" in spec:
-            config_args.append("--with-pydebug")
-        else:
-            config_args.append("--without-pydebug")
-
-        if "+shared" in spec:
-            config_args.append("--enable-shared")
-        else:
-            config_args.append("--disable-shared")
-
-        config_args.append("--without-ensurepip")
-
-        if "+pic" in spec:
-            cflags.append(self.compiler.cc_pic_flag)
-
-        if "+ssl" in spec:
-            config_args.append("--with-openssl={0}".format(spec["openssl"].prefix))
+                path = os.path.join(self.prefix, "python{0}.exe".format(ver))
+            if os.path.exists(path):
+                return Executable(path)
 
-        if "+dbm" in spec:
-            # Default order is ndbm:gdbm:bdb
-            config_args.append("--with-dbmliborder=gdbm")
         else:
-            config_args.append("--with-dbmliborder=")
+            msg = "Unable to locate {0} command in {1}"
+            raise RuntimeError(msg.format(self.name, self.prefix.bin))
 
-        if "+pyexpat" in spec:
-            config_args.append("--with-system-expat")
-        else:
-            config_args.append("--without-system-expat")
+    @property
+    def config_vars(self):
+        """Return a set of variable definitions associated with a Python installation.
 
-        if "+ctypes" in spec:
-            config_args.append("--with-system-ffi")
-        else:
-            config_args.append("--without-system-ffi")
+        Wrapper around various ``sysconfig`` functions. To see these variables on the
+        command line, run:
 
-        if "+tkinter" in spec:
-            config_args.extend(
-                [
-                    "--with-tcltk-includes=-I{0} -I{1}".format(
-                        spec["tcl"].prefix.include, spec["tk"].prefix.include
-                    ),
-                    "--with-tcltk-libs={0} {1}".format(
-                        spec["tcl"].libs.ld_flags, spec["tk"].libs.ld_flags
-                    ),
-                ]
-            )
+        .. code-block:: console
```

Comment:
The code block below:
```
# https://docs.python.org/3.8/library/sqlite3.html#f1
if spec.satisfies("+sqlite3 ^sqlite+dynamic_extensions"):
    config_args.append("--enable-loadable-sqlite-extensions")
```
is completely unrelated. Should have been:
```
python -m sysconfig
```

Reply:
Confirmed that the `configure_args` have this stanza.